### PR TITLE
Add secure random to LazySingletons

### PR DESCRIPTION
**Pull request by DomGarguilo**
A new class, `LazySingletons` was added in #3531 which hosts singleton suppliers. This PR adds a supplier to that class for a `SecureRandom` object and replaces the creation of new `SecureRandom` objects with that supplier.

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/util/LazySingletons.java [hunk 1]
```diff
@@ -37,4 +38,6 @@ private LazySingletons() {}
    */
   public static final Supplier<Gson> GSON = Suppliers.memoize(Gson::new);
 
+  public static final Supplier<SecureRandom> SECURE_RANDOM = Suppliers.memoize(SecureRandom::new);
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/util/LazySingletons.java:41]:
```suggestion
  /**
   * A SecureRandom instance created with the default constructor.
   */
  public static final Supplier<SecureRandom> SECURE_RANDOM = Suppliers.memoize(SecureRandom::new);
```

**Review by ctubbsii — COMMENTED** [review]
This is fine, but one thing that might help this be used more cleanly is using a shorter name.

Here are some I was considering `RNG`, `SRNG`, `SRAND`, `RAND`, `SECRAND`, or `RANDOM`.

I think of these, I like `RNG` the best, as it's the most succinct, or `SRAND` as the most concise expression that it's a SecureRandom vs. a regular Random. The fact that it's a SecureRandom vs. just a Random is expressed in the type, either way, so I don't think putting the `SECURE` in the name (or even just `S`) helps make the calling code any more readable over `RNG`.

**Comment by DomGarguilo** [comment]
> The fact that it's a SecureRandom vs. just a Random is expressed in the type, either way, so I don't think putting the `SECURE` in the name (or even just `S`) helps make the calling code any more readable over `RNG`

I agree. I think that `RANDOM` and `RNG` are my top choices for renaming but I'm leaning towards `RANDOM` just so it is a bit clearer what the variables type is. Also it seems like when we created `SecureRandom` objects throughout the code, we often use "random" as the variable name so it would be more consistent with what is already present.

**Comment by ctubbsii** [comment]
> > The fact that it's a SecureRandom vs. just a Random is expressed in the type, either way, so I don't think putting the `SECURE` in the name (or even just `S`) helps make the calling code any more readable over `RNG`
> 
> I agree. I think that `RANDOM` and `RNG` are my top choices for renaming but I'm leaning towards `RANDOM` just so it is a bit clearer what the variables type is. Also it seems like when we created `SecureRandom` objects throughout the code, we often use "random" as the variable name so it would be more consistent with what is already present.

I'm fine with `RANDOM`, but if you wanted to, you could also go with `SRANDOM`, just to make it super clear. While the local uses may have not cared what type of random it was (they were probably just using SecureRandom to avoid a spotbugs warning), we have to also consider how the LazySingletons class presents itself, even when the callers don't necessarily care.

**Review by ctubbsii — APPROVED** [review]
Feel free to merge whenever you've got the name you think works the best.

**Pull request merged**